SSA/Projects/LLVMAndRiscv.lean

```diff
@@ -0,0 +1,144 @@
+import SSA.Core.MLIRSyntax.EDSL
+import SSA.Projects.InstCombine.LLVM.PrettyEDSL
+import SSA.Core.Framework
+import SSA.Core.Util
+import SSA.Core.Util.ConcreteOrMVar
+import SSA.Projects.InstCombine.ForStd
+import SSA.Projects.InstCombine.LLVM.Semantics
+import SSA.Projects.InstCombine.Tactic
+import SSA.Projects.RISCV64.Syntax
+import SSA.Projects.RISCV64.Base
+import SSA.Projects.RISCV64.Semantics
+import SSA.Projects.RISCV64.PrettyEDSL
+import SSA.Core.HVector
+
+open InstCombine(LLVM)
+namespace LLVMRiscV
+
+
+/-! # Hybrid dialect -/
+/-
+This file contains a hybrid dialect combining
+SSA.Projects.RISCV64 and SSA.Projects.InstCombine.
+Modelling two dialects within a larger hybrid dialect allows us
+to reuse existing infrastructure such as the Peephole Rewriter which
+currently only operates within one dialect.
+
+To make the intermixing of the type system across the dialects work,
+we insert unrealized_conversion_cast like MLIR does during lowering.
+see: section (UnrealizedConversionCastOp)
+https://mlir.llvm.org/docs/Dialects/Builtin/#builtinunrealized_conversion_cast-unrealizedconversioncastop
+ -/
+
+/-- Defining an instance for LLVM.Ty from InstCombine.Ty instance.-/
+instance : DecidableEq LLVM.Ty :=
+  fun a b => by
+    simp only [LLVM] at a b
+    exact (inferInstanceAs (DecidableEq (InstCombine.MTy 0)) a b)
+
+/-- Defining an instance for LLVM.Op from InstCombine.Op instance. -/
+instance : DecidableEq LLVM.Op :=
+  fun a b => by
+    simp only [LLVM] at a b
+    exact (inferInstanceAs (DecidableEq (InstCombine.MOp 0)) a b)
+
+inductive Ty where
+  | llvm : (LLVM.Ty) -> Ty
+  | riscv : (Dialect.Ty RISCV64.RV64) -> Ty
```

Comment:
```suggestion
  | riscv : RISCV64.RV64.Ty -> Ty
```

Even though the dot notation is maybe slightly mis-leading, it does look nicer and we do use it in other files, so it's best to be consistent.